versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/DatabaseAdapterExtension.java

```diff
@@ -0,0 +1,190 @@
+      TestConfigurer testConfigurer,
+      DatabaseAdapterFactory.Builder<C> adapterBuilder,
+      Function<C, DatabaseAdapterConfig> configurer) {
+    adapterBuilder =
+        adapterBuilder
+            .configure(SystemPropertiesConfigurer::configureFromSystemProperties)
+            // default to a quite small max-size for the CommitLogEntry.keyList + KeyListEntity
+            // This is necessary for AbstractManyKeys to work properly!!
+            .configure(c -> c.withMaxKeyListSize(2048))
+            .configure(configurer)
+            .configure(testConfigurer::configure);
+
+    return adapterBuilder.build();
+  }
+
+  static class AdapterResource implements CloseableResource {
+    private final DatabaseAdapter databaseAdapter;
+
+    AdapterResource(DatabaseAdapter databaseAdapter) {
+      this.databaseAdapter = databaseAdapter;
+    }
+
+    @Override
+    public void close() throws Throwable {
+      databaseAdapter.close();
+    }
+  }
+
+  @Override
+  public void beforeAll(ExtensionContext context) throws Exception {
+    Class<?> testClass = context.getRequiredTestClass();
+    findAnnotatedFields(testClass, Adapter.class, ReflectionUtils::isStatic)
+        .forEach(
+            field -> {
+              assertValidStaticFieldCandidate(field);
+              Adapter annotation =
+                  findAnnotation(field, Adapter.class).orElseThrow(IllegalStateException::new);
+              try {
+                if (context.getStore(NAMESPACE).get(KEY) != null) {
+                  throw new JUnitException(
+                      "Only one instance of '@DatabaseAdapterExtension.Adapter static DatabaseAdapter databaseAdapter;'"
+                          + "field allowed per test class");
+                }
+
+                TestConfigurer configurer =
+                    makeAccessible(annotation.configurer().getDeclaredConstructor()).newInstance();
+                DatabaseAdapter adapter =
+                    context
+                        .getStore(NAMESPACE)
+                        .getOrComputeIfAbsent(
+                            KEY,
+                            key -> new AdapterResource(createAdapter(context, configurer)),
+                            AdapterResource.class)
+                        .databaseAdapter;
+
+                makeAccessible(field).set(null, adapter);
+              } catch (Throwable t) {
+                ExceptionUtils.throwAsUncheckedException(t);
+              }
+            });
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext context) {
+    AdapterResource adapter = context.getStore(NAMESPACE).get(KEY, AdapterResource.class);
+
+    if (adapter == null) {
+      throw new JUnitException(
+          "Test class does not declare mandatory '@DatabaseAdapterExtension.Adapter static DatabaseAdapter databaseAdapter;' field");
+    }
+
+    adapter.databaseAdapter.reinitializeRepo("main");
```

Comment:
I'd keep this part in a `@BeforeEach` method on `AbstractTieredCommitsTest`. it kind of belongs more with the actual testing logic, while "extensions" can concentrate of the storage and adapter setup part.